index.js

```diff
@@ -3,57 +3,77 @@
 const fp = require('fastify-plugin')
 const formatEntry = require('./lib/formatEntry')
 const CRLF = '\r\n'
+const EarlyHints = `HTTP/1.1 103 Early Hint${CRLF}`
 
 function fastifyEarlyHints (fastify, opts, next) {
   if (fastify.initialConfig.http2 === true) {
-    return next(new Error('Early Hints cannot be used with a HTTP2 server.'))
+    return next(Error('Early Hints cannot be used with a HTTP2 server.'))
   }
 
   const formatEntryOpts = {
     warn: opts.warn
   }
 
-  function earlyHints (reply) {
-    const promiseBuffer = []
-    const serialize = function (c) {
-      let message = `HTTP/1.1 103 Early Hints${CRLF}`
-      for (let i = 0; i < c.length; i++) {
-        message += `${formatEntry(c[i], formatEntryOpts)}${CRLF}`
-      }
-      return `${message}${CRLF}`
-    }
-
-    return {
-      close: async function () {
-        if (promiseBuffer.length) {
-          await Promise.all(promiseBuffer)
+  fastify.decorateReply('writeEarlyHints', function (headers) {
```

Comment:
Could we move this function to the upper scope?

Reply:
Does it matter? It only initialize once.

Reply:
just 2 indentation spaces less